docs/aws/audit/elbmonitoring/rules/elb_connection_draining.mdx

```diff
@@ -24,6 +24,223 @@ NIST, NISTCSF
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start using AWS CLI, you need to install it on your local system and configure it with your AWS account credentials. You can do this by running the following commands:
+
+   Installation:
+   ```
+   pip install awscli
+   ```
+   Configuration:
+   ```
+   aws configure
+   ```
+   You will be prompted to provide your AWS Access Key ID, Secret Access Key, Default region name, and Default output format.
+
+2. List all the Load Balancers: Use the following command to list all the load balancers in your AWS account:
+
+   ```
+   aws elbv2 describe-load-balancers
+   ```
+   This command will return a JSON output with details of all the load balancers.
+
+3. Extract Load Balancer ARNs: From the JSON output, extract the ARN of each load balancer. You can use the following command to do this:
+
+   ```
+   aws elbv2 describe-load-balancers --query 'LoadBalancers[*].[LoadBalancerArn]' --output text
+   ```
+   This command will return a list of all the load balancer ARNs.
+
+4. Check Connection Draining: For each load balancer ARN, use the following command to check if connection draining is enabled:
+
+   ```
+   aws elbv2 describe-load-balancer-attributes --load-balancer-arn <LoadBalancerArn>
+   ```
+   Replace `<LoadBalancerArn>` with the ARN of the load balancer you want to check. This command will return a JSON output with the attributes of the load balancer. If connection draining is enabled, the `ConnectionDraining.Enabled` attribute will be `true`. If it's not, then the load balancer is misconfigured.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary AWS SDK for Python (Boto3) if you haven't done so already. You can install it using pip:
+
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and create a session using your AWS credentials:
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='YOUR_REGION'
+)
+```
+
+3. Create an ELB client and retrieve all the load balancers:
+
+```python
+elb = session.client('elb')
+
+load_balancers = elb.describe_load_balancers()
+```
+
+4. Iterate over the load balancers and check if connection draining is enabled:
+
+```python
+for lb in load_balancers['LoadBalancerDescriptions']:
+    attributes = elb.describe_load_balancer_attributes(LoadBalancerName=lb['LoadBalancerName'])
+    if 'ConnectionDraining' in attributes['LoadBalancerAttributes']:
+        if not attributes['LoadBalancerAttributes']['ConnectionDraining']['Enabled']:
+            print(f"Connection Draining is not enabled for {lb['LoadBalancerName']}")
+```
+
+This script will print the names of all load balancers that do not have connection draining enabled.
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>